addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -236,31 +288,36 @@ options.registry.gallery = options.Class.extend({
         }
         // TODO Remove in master.
         // Order might be wrong if images were not loaded yet.
-        while (imgs.length) {
+        let index = 0;
+        while (mediaHolderEls.length) {
             var min = Infinity;
             var $lowest;
             _.each(cols, function (col) {
                 var $col = $(col);
-                var height = $col.is(':empty') ? 0 : $col.find('img').last().offset().top + $col.find('img').last().height() - self.$target.offset().top;
+                const colMediaEls = self._getAllMediaEls(col);
+                let height = 0;
+                if (colMediaEls.length) {
+                    const lastMediaRect = colMediaEls[colMediaEls.length - 1].getBoundingClientRect();
+                    height = lastMediaRect.top + lastMediaRect.height - self.$target[0].getBoundingClientRect().top;
+                }
                 // Neutralize invisible sub-pixel height differences.
                 height = Math.round(height);
                 if (height < min) {
                     min = height;
                     $lowest = $col;
                 }
             });
-            // Only on Chrome: appended images are sometimes invisible and not
-            // correctly loaded from cache, we use a clone of the image to force
-            // the loading.
-            $lowest.append(imgs.shift().cloneNode(true));
```

Comment:
Note: during conflict resolution, there are 3-4 places where you changed a `img` into a `mediaHolderEl` but where in 16.0 a `.cloneNode(true)` was added (see here with a comment explaining why). During conflict resolution, I just ignored that and kept your version with `mediaHolderEl` but probably worth checking if any issue due to not cloning :pray: